src/utils/model_library.py

```
@@ -0,0 +1,627 @@
+                        dir=model_directory,
+                        prefix=f".{slot}-{generation}-",
+                        suffix=".tmp",
+                    )
+                    temporary_path = Path(temporary_name)
+                    temporary_paths.append(temporary_path)
+                    with os.fdopen(descriptor, "wb") as output_stream:
+                        with source_path.open("rb") as input_stream:
+                            shutil.copyfileobj(input_stream, output_stream)
+                        output_stream.flush()
+                        os.fsync(output_stream.fileno())
+                for slot, temporary_path in zip(new_paths, temporary_paths):
```

Ruff (0.16.0)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)